zutil_p.h

```diff
@@ -19,12 +19,14 @@ static inline void *zng_alloc(size_t size) {
 #ifdef HAVE_POSIX_MEMALIGN
     void *ptr;
     return posix_memalign(&ptr, 64, size) ? NULL : ptr;
+#elif defined(HAVE_ALIGNED_ALLOC)
+    return (void *)aligned_alloc(64, size);  /* Defined in C11 */
 #elif defined(_WIN32)
     return (void *)_aligned_malloc(size, 64);
 #elif defined(__APPLE__)
-    return (void *)malloc(size);     /* MacOS always aligns to 16 bytes */
-#elif defined(HAVE_ALIGNED_ALLOC)
-    return (void *)aligned_alloc(64, size);
+    /* Fallback for when posix_memalign and aligned_alloc are not available. 
+     * On macOS, it always aligns to 16 bytes. */
```

Comment:
I'm not sure if people try to compile zlib-ng with older versions than OS X, but changing the capitalization also change the underlying meaning of the comment. With capital M, it meant any version of the operating system, not just the latest versions.

Reply:
I don't think most people would read that much into the capitalization of the OS name.

Reply:
Actually, I think it might be better to prefer `aligned_alloc()` as the first choice and `posix_memalign()` as second choice.
Not because there is anything wrong with the latter one, but the former is after all part of the C11 standard now, and zlib-ng 2.1.x has C11 as a minimum requirement.

Interestingly windows does not seem to support it, ref: [C11 arriving in MSVC](https://devblogs.microsoft.com/cppblog/c11-and-c17-standard-support-arriving-in-msvc/)
`Due to the nature of the Windows heap, aligned_alloc support is missing. The alternative is to use _aligned_malloc`
Makes me wonder what function we use on windows..

Reply:
> Makes me wonder what function we use on windows..

We use `_aligned_malloc` on Windows.

Reply:
I have updated it so `HAVE_ALIGNED_ALLOC` is very first.

Reply:
Yes, I've always hated the fact that Microsoft never cared to write an allocator that handed out aligned heap allocations that could be freed by normal free.  This causes issues when you expect foreign libraries to take ownership over your allocations and basically forces you to provide a "free" wrapper.  

My guess is on Windows we are using our fallback that just over allocates and returns a pointer to the allocated region.  Windows _does_ have _mm_malloc, likely due to Intel insisting its existence very early on, but unlike the glibc versions of this, it can't be freed by normal "free", it needs "_mm_free".

Reply:
> We use `_aligned_malloc` on Windows.

I skipped right over the `_WIN32` section, didn't I

> I have updated it so HAVE_ALIGNED_ALLOC is very first.

LGTM